File: ios/SimpleModule.m
//
//  SimpleModule.m
//  CozyAI
//
//  Created by God of Code on 5/13/25.
//

#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(SimpleModule, NSObject)
RCT_EXTERN_METHOD(greet:(NSString *)name
                  resolver:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)
@end

File: ios/SimpleModule.swift
//
//  SimpleModule.swift
//  CozyAI
//
//  Created by God of Code on 5/13/25.
//

import Foundation
import React

@objc(SimpleModule)
class SimpleModule: NSObject {
  @objc
  func greet(_ name: String, resolver resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
    let message = "Привет, \(name)!"
    resolve(message)
  }

  @objc
  static func requiresMainQueueSetup() -> Bool {
    return false
  }
}
